pycbc/events/status.py

```diff
@@ -0,0 +1,252 @@
+                           veto_definer=None):
+    """Return the times where any flag is active 
+
+    Parameters
+    ----------
+    ifo: string
+        The interferometer to query (H1, L1).
+    segment_name: list of strings
+        The status flag to query from LOSC.
+    start_time: int 
+        The starting gps time to begin querying from LOSC
+    end_time: int 
+        The end gps time of the query
+    source: str, Optional
+        Choice between "GWOSC" or "dqsegdb". If dqsegdb, the server option may
+        also be given. The default is to try GWOSC first then try dqsegdb.
+    server: str, Optional
+        The server path. Only used with dqsegdb atm.
+    veto_definer: str, Optional
+        The path to a veto definer to define groups of flags which
+        themselves define a set of segments.
+
+    Returns
+    ---------
+    segments: glue.segments.segmentlist
+        List of segments
+    """
+    total_segs = None
+    for flag_name in segment_names:
+        segs = query_flag(ifo, flag_name, start_time, end_time,                                       
+                          source=source, server=server,
+                          veto_definer=veto_definer)
+        if total_segs is not None:
+            total_segs = (total_segs + segs).coalesce()
+        else:
+            total_segs = segs
+    return total_segs
+
+def query_combined_flags(ifo, up_flags, start_time, end_time, down_flags=None,               
+                        source='any', server="segments.ligo.org",
+                        veto_definer=None):
+    """Return the times where any "up" flag is active minus "down" flag times
+
+    Parameters
+    ----------
+    ifo: string
+        The interferometer to query (H1, L1).
+    up flags: list of strings
+        The status flag to query from LOSC.
+    start_time: int 
+        The starting gps time to begin querying from LOSC
+    end_time: int 
+        The end gps time of the query
+    down flags: list of strings
+        Flags which indicate times to subtract from the combined segments
+    source: str, Optional
+        Choice between "GWOSC" or "dqsegdb". If dqsegdb, the server option may
+        also be given. The default is to try GWOSC first then try dqsegdb.
+    server: str, Optional
+        The server path. Only used with dqsegdb atm.
+    veto_definer: str, Optional
+        The path to a veto definer to define groups of flags which
+        themselves define a set of segments.
+
+    Returns
+    ---------
+    segments: glue.segments.segmentlist
+        List of segments
+    """
+    segments = query_cumulative_flags(ifo, up_flags, start_time, end_time,
+                                      source=source, server=server,
+                                      veto_definer=veto_definer)
+    down_flags = [] if down_flags is None else down_flags
+    for flag_name in down_flags:
+        mseg = query_flag(ifo, flag_name, start_time, end_time)
```

Comment:
I'm unclear why this doesn't also use `query_cumulative_flags`.

.... This seems to allow for there to be some uncertainties here. For example if I do:

```science=SCIENCE-CAT_1+MISLABELED_CAT_1_I_ACTUALLY_WANT_TO_ANALYSE```

I think I would not get `MISLABELED_CAT_1_I_ACTUALLY_WANT_TO_ANALYSE`. In other words if a segment is added *and* subtracted in multiple places the behaviour seems ill defined.

Reply:
Yes, it is ill defined then. I've added to the documentation that positive time-adding flags must precede negations.